torch_frame/datasets/huggingface_dataset.py

```diff
@@ -0,0 +1,139 @@
+
+    Args:
+        path (str): Path or name of the dataset.
+        name (str, optional): Defining the name of the dataset configuration.
+
+    Example:
+        Load the `spotify-tracks-dataset` dataset from the Hugging Face Hub
+        to the :obj:`torch_frame.data.Dataset`:
+
+    .. code-block:: python
+
+        >>> import torch_frame
+        >>> from torch_frame.datasets import HuggingFaceDatasetDict
+        >>> from torch_frame.config.text_embedder import TextEmbedderConfig
+        >>> from torch_frame.testing.text_embedder import HashTextEmbedder
+        >>> dataset = HuggingFaceDatasetDict(
+        ...     path="maharshipandya/spotify-tracks-dataset",
+        ...     col_to_stype = {
+        ...         "artists": torch_frame.categorical,
+        ...         "album_name": torch_frame.text_embedded,
+        ...         "track_name": torch_frame.text_embedded,
+        ...         "popularity": torch_frame.numerical,
+        ...         "duration_ms": torch_frame.numerical,
+        ...         "explicit": torch_frame.categorical,
+        ...         "danceability": torch_frame.numerical,
+        ...         "energy": torch_frame.numerical,
+        ...         "key": torch_frame.categorical,
+        ...         "loudness": torch_frame.numerical,
+        ...         "mode": torch_frame.categorical,
+        ...         "speechiness": torch_frame.numerical,
+        ...         "acousticness": torch_frame.numerical,
+        ...         "instrumentalness": torch_frame.numerical,
+        ...         "liveness": torch_frame.numerical,
+        ...         "valence": torch_frame.numerical,
+        ...         "tempo": torch_frame.numerical,
+        ...         "time_signature": torch_frame.categorical,
+        ...         "track_genre": torch_frame.categorical,
+        ...     },
+        ...     target_col="track_genre",
+        ...     col_to_text_embedder_cfg=TextEmbedderConfig(
+        ...         text_embedder=HashTextEmbedder(10)),
+        ... )
+        >>> dataset.materialize()
+        >>> dataset.tensor_frame
+        TensorFrame(
+            num_cols=18,
+            num_rows=114000,
+            categorical (5): [
+                'artists',
+                'explicit',
+                'key',
+                'mode',
+                'time_signature',
+            ],
+            numerical (11): [
+                'acousticness',
+                'danceability',
+                'duration_ms',
+                'energy',
+                'instrumentalness',
+                'liveness',
+                'loudness',
+                'popularity',
+                'speechiness',
+                'tempo',
+                'valence',
+            ],
+            embedding (2): ['album_name', 'track_name'],
+            has_target=True,
+            device='cpu',
+        )
+
+    """
+    def __init__(
+        self,
+        path: str,
+        col_to_stype: dict[str, stype],
```

Comment:
Can we make this optional and call the `infer_df_stype` internally? Otherwise, it's too tedious to specify stype for each column.